src/storage/copier/column_chunk.cpp

```diff
@@ -17,8 +17,19 @@ ColumnChunk::ColumnChunk(LogicalType dataType, CopyDescription* copyDescription,
 
 ColumnChunk::ColumnChunk(
     LogicalType dataType, offset_t numValues, CopyDescription* copyDescription, bool hasNullChunk)
-    : dataType{std::move(dataType)}, numBytesPerValue{getDataTypeSizeInChunk(this->dataType)},
-      numBytes{numBytesPerValue * numValues}, copyDescription{copyDescription} {
+    : dataType{std::move(dataType)}, numBitsPerValue{getDataTypeSizeInChunk(this->dataType)},
+      numBytesPerValue{numBitsPerValue / 8}, numBytes{numBitsPerValue * numValues / 8},
+      copyDescription{copyDescription} {
+    // TODO(bmwinger): Move logic into NullColumnChunk
```

Comment:
I agree, moving this logic into NullColumnChunk is a better idea. We should also move `numBitsPerValue` to be a private field of NullColumnChunk.
You can add a virtual `initialize` function, which is responsible for setting `numBytesPerValue`, `numBytes`, and allocate `buffer` in ColumnChunk. NullColumnChunk will override this function accordingly.